apps/src/lab2/responseValidators.ts

```diff
@@ -1,26 +1,45 @@
 import {ResponseValidator} from '@cdo/apps/util/HttpClient';
-import {BlocklySource, ProjectSources} from './types';
+import {BlocklySource, LevelProperties, ProjectSources} from './types';
 
 export const SourceResponseValidator: ResponseValidator<
   ProjectSources
 > = response => {
-  const projectSources = response as ProjectSources;
-  if (!projectSources.source) {
+  if (!response.source) {
     throw new ValidationError('Missing required field: source');
   }
 
   // Currently only Blockly JSON sources are supported.
   let blocklySource;
   try {
-    blocklySource = JSON.parse(projectSources.source) as BlocklySource;
+    blocklySource = JSON.parse(response.source as string) as BlocklySource;
   } catch (e) {
     throw new ValidationError('Error parsing JSON: ' + e);
   }
   if (blocklySource.blocks === undefined) {
     throw new ValidationError('Missing required field: blocks');
   }
 
-  return projectSources;
+  return response as unknown as ProjectSources;
+};
+
+export const LevelPropertiesValidator: ResponseValidator<
```

Comment:
Do you recommend that we add a `ChatCompletionValidator` for the response returned from the Open AI chat completion controller? Just curious.

Reply:
We could, yeah. It's not necessary to call the API, but if we want to be stricter about the type we're getting back from the server it could be good to have. That being said, since we make a POST request, we'll probably have to update the HttpClient to accept a validator since it doesn't currently for post requests. Alternatively, we can just perform our own validation after we get the response back.